visualization/react-lib/src/components/resource-estimator/TimeDiagram.tsx

```diff
@@ -0,0 +1,206 @@
+import React from "react";
+import { TableComponent, IItem, IState } from "../table/Table";
+import { ThemeProvider, IGroup, IColumn } from "@fluentui/react";
+import { JobResults } from "../../models/JobResults";
+import "./Diagram.css";
+import { TimeChart } from "../time-chart";
+import { GetColumns } from "../table/Column";
+
+interface TimeDiagramProps {
+  data: string;
+}
+
+function TimeDiagram({ data }: TimeDiagramProps) {
+  // Parse job results data.
+  let jobResults = JSON.parse(data) as JobResults;
+
+  /*------------------------------ Configure canvas sizing ------------------------------  */
+  const diagramRef = React.useRef<any>();
+  const [width, setWidth] = React.useState(0);
+  const [height, setHeight] = React.useState(0);
+
+  const handleWidth = () => {
+    const width = diagramRef?.current?.offsetWidth;
+    if (width) {
+      setWidth(width);
+    }
+  }
+  const handleSize = () => {
+    handleWidth();
+    const height = diagramRef?.current?.offsetHeight;
+    if (height) {
+      setHeight(height);
+    }
+  }
+  React.useLayoutEffect(() => {
+    handleSize();
+    window.addEventListener("resize", handleWidth);
+  }, [diagramRef.current]);
+
+
+  /*------------------------------  Define and parse table data ------------------------------  */
+  const algorithmRuntimeFormatted = jobResults.physicalCountsFormatted.runtime;
+  const tFactoryRuntimeFormatted =
+    jobResults.physicalCountsFormatted.tfactoryRuntime;
+  const numTFactoryInvocations =
+    jobResults.physicalCounts.breakdown.numTfactoryRuns;
+
+  const numTfactories = jobResults.physicalCounts.breakdown.numTfactories;
+  const numTStatesPerSingleTfactory = jobResults.tfactory.numTstates;
+  const numTStatesAllTfactoriesOneInvocation = numTStatesPerSingleTfactory * numTfactories;
+
+  const numTStatesPerInvocationString = "Output T states of single T factory  (" + numTStatesPerSingleTfactory + ") * T factories (" + numTfactories + ") = " + numTStatesAllTfactoriesOneInvocation
+    + " T states produced by a single invocation of all T factories."
+  const tableItems: IItem[] = [
+    {
+      name: "Algorithm runtime",
+      value: algorithmRuntimeFormatted,
+      description: "Total runtime of algorithm."
+    },
+    {
+      name: "T factory runtime",
+      value: tFactoryRuntimeFormatted,
+      description: "Runtime of a single T factory."
+    },
+    {
+      name: "T factory copies",
+      value: numTfactories.toLocaleString(),
+      description: "Number of T factories executed in parallel capable of producing the demanded T states during the algorithm's runtime."
+    },
+    {
+      name: "T factory invocations",
+      value: numTFactoryInvocations.toLocaleString(),
+      description: "Number of times all T factories are invoked concurrently.",
+    },
+    {
+      name: "T states per single T factory run",
+      value: "~ " + numTStatesPerSingleTfactory,
```

Comment:
Why is there a ~? It should be `numTStatesPerSingleTfactory`

Reply:
Hmm originally I think I thought that this was a number that was rounded in the results, but I will remove the ~